src/libs/tmdk/tests/deploy.test.ts

```diff
@@ -1,26 +1,265 @@
+  CreateSceneCommand,
+  GetComponentTypeCommand,
+  GetEntityCommand,
+  GetWorkspaceCommand,
+  IoTTwinMakerClient,
+  ResourceNotFoundException,
+} from "@aws-sdk/client-iottwinmaker";
+import {
+  emptyTmdk,
+  componentType1Definition,
+  scene1,
+  entity1Definition,
+} from "./test-constants";
+import { PutObjectCommand, S3Client } from "@aws-sdk/client-s3";
+import * as fsPromises from "fs/promises";
+const fs = require("fs");
 
-jest.mock("../src/lib/aws-clients");
+jest.mock("fs/promises", () => {
+  return { readFile: jest.fn() };
+});
+const twinmakerMock = mockClient(IoTTwinMakerClient);
+const s3Mock = mockClient(S3Client);
+const fakeTmdkDir = "/tmp/deploy-unit-tests";
 
 describe("testing deploy", () => {
-  beforeEach(mockExit.mockClear);
+  beforeEach(() => {
+    twinmakerMock.reset();
+    s3Mock.reset();
+    jest.resetAllMocks();
+    mockExit.mockClear();
+  });
 
   test("deploy_givenNoTMDKProject_expectError", async () => {
     const argv2 = {
       _: ["deploy"],
       $0: "tmdk_local",
       region: "us-east-1",
       "workspace-id": "irrelevant",
-      dir: "/tmp/deploy-unit-tests/i-do-not-exist",
+      dir: "i-do-not-exist",
+    } as Arguments<Options>;
+    await expect(handler(argv2)).rejects.toThrow(
+      Error("TDMK.json does not exist. Please run tmdk init first.")
+    );
+  });
+
+  test("deploy_givenNoWorkspace_expectError", async () => {
+    fs.existsSync = jest.fn(() => {
+      return true;
+    });
+    jest.spyOn(fs, "readFileSync").mockReturnValue(JSON.stringify({}, null, 4));
+    twinmakerMock
+      .on(GetWorkspaceCommand)
+      .rejects(new ResourceNotFoundException({ $metadata: {}, message: "" }));
+
+    const argv2 = {
+      _: ["init"],
+      $0: "tmdk_local",
+      region: "us-east-1",
+      "workspace-id": "non-existent",
+      dir: fakeTmdkDir,
+    } as Arguments<Options>;
+    await expect(handler(argv2)).rejects.toThrow(ResourceNotFoundException);
+  });
+
+  test("deploy_givenEmptyProject_expectSuccess", async () => {
+    fs.existsSync = jest.fn(() => {
+      return true;
+    });
+    fs.readFileSync = jest.fn((path: [string]) => {
+      if (path.includes("tmdk")) {
+        return JSON.stringify(emptyTmdk, null, 4);
+      } else {
+        // covers entities.json
+        return JSON.stringify([], null, 4);
+      }
+    });
+    twinmakerMock.on(GetWorkspaceCommand).resolves({});
+
+    const argv2 = {
+      _: ["init"],
+      $0: "tmdk_local",
+      region: "us-east-1",
+      "workspace-id": workspaceId,
+      dir: fakeTmdkDir,
+    } as Arguments<Options>;
+    expect(await handler(argv2)).toBe(0);
+    expect(twinmakerMock.commandCalls(CreateComponentTypeCommand).length).toBe(
+      0
+    );
+    expect(twinmakerMock.commandCalls(CreateEntityCommand).length).toBe(0);
+    expect(twinmakerMock.commandCalls(CreateSceneCommand).length).toBe(0);
+  });
+
+  test("deploy_given1ComponentType_expectSuccess", async () => {
+    fs.existsSync = jest.fn(() => {
```

Comment:
() => false

Reply:
Got it, did not know that would work. Thanks!